src/algorithms/calorimetry/CalorimeterHitsMerger.cc

```diff
@@ -140,4 +178,44 @@ void CalorimeterHitsMerger::process(
     debug("Size before = {}, after = {}", in_hits->size(), out_hits->size());
 }
 
+void CalorimeterHitsMerger::build_map_via_funcs(
+  const edm4eic::CalorimeterHitCollection* in_hits,
+  MergeMap& merge_map
+) const {
+
+  std::vector<RefField> ref_fields;
+  for (std::size_t iHit = 0; const auto& hit : *in_hits) {
+
+    // make sure vector is clear
+    ref_fields.clear();
+    for (std::size_t iField = 0; const auto& name_field : id_desc.fields()) {
+
+      // check if field has associated mapping
+      const bool foundMapping = (
```

Comment:
This looks a bit complicated, and I doubt that it actually works for mixing of "mappings" transformations and "refs" bitmasks. Could it be simplified assuming 1-to-1 loop?

Reply:
Yes, as-is it's set up so that if a user provides a mapping, it defaults to using that and will ignore any provided bitmasks. But it can definitely be just turned into a 1-to-1 loop where it checks for the mapping or the bitmask.